src/Gaffer/MetadataAlgo.cpp

```diff
@@ -177,6 +178,28 @@ bool affectedByChange( const Node *node, IECore::TypeId changedNodeTypeId, const
 	return node->isInstanceOf( changedNodeTypeId );
 }
 
+void copy( const GraphComponent *from, GraphComponent *to, const StringAlgo::MatchPattern &exclude, bool persistentOnly, bool persistent )
```

Comment:
@ivanimanishi no need for you to review the entire PR, but it might be worth comparing this implementation to your Jabuka version to see if there's any other magic you added there that could fit here.